### PR TITLE
dashboard and tutorials data, models and labels locations (issue #667)

### PR DESCRIPTION
**Pull request by elboyran**
solves issue #667

### CODE REVIEW

**Comment by elboyran** [comment]
Struggle to make the linter happy :-(
Also, how to test the dashboard in this branch (it's disabled and I struggle to find out how to enable it)?
@cpranav93 , @cwmeijer any ideas?

**Comment by elboyran** [comment]
Conflicting files to be resolved before merging.

**Comment by elboyran** [comment]
Old and proposed folder structures of `dianna/dianna `and `dianna/tutorials` :
---------------------
## `main` branch
### `dianna` (6 directories, 56 files)

├── __init__.py
├── ...
├── dashboard
│   ├── ...
├── data
│   ├── bee_2.png
│   ├── binary-mnist.npz
│   ├── digit0.jpeg
│   ├── digit0.jpg
│   ├── digit0.png
│   ├── digit1.png
│   ├── labels_mnist.txt
│   ├── labels_resnet50.txt
│   ├── labels_text.txt
│   ├── leafsnap_model.onnx
│   ├── logo.png
│   ├── mnist_model_tf.onnx
│   ├── movie_review_model.onnx
│   ├── movie_reviews_word_vectors.txt
│   ├── season_prediction_model_temp_max_binary.onnx
│   ├── weather_data.npy
│   └── weather_data_labels.txt
├── methods
│   ├── ...
├── utils
│   ├── ...
└── visualization
│   ├── ...

### `tutorials` (4 directories, 38 files)

├── README.md
├── conversion_onnx
│   ├── ...
├── data
│   ├── binary-mnist.npz
│   ├── coffee_test.csv
│   ├── coffee_train.csv
│   ├── leafsnap_classes.csv
│   ├── leafsnap_example_acer_rubrum.jpg
│   ├── movie_reviews_word_vectors.txt
│   └── shapes.npz
├── demo.ipynb
├── img
│   ├── bee.jpg
│   └── digit0.jpg
├── kernelshap_geometric_shapes.ipynb
├── kernelshap_mnist.ipynb
├── kernelshap_tabular_penguin.ipynb
├── kernelshap_tabular_weather.ipynb
├── lime_images.ipynb
├── lime_tabular_penguin.ipynb
├── lime_tabular_weather.ipynb
├── lime_text.ipynb
├── lime_timeseries_coffee.ipynb
├── lime_timeseries_weather.ipynb
├── models
│   ├── coffee.onnx
│   ├── geometric_shapes_model.onnx
│   ├── leafsnap_model.onnx
│   ├── mnist_model.onnx
│   ├── mnist_model_tf.onnx
│   ├── movie_review_model.onnx
│   ├── penguin_model.onnx
│   ├── season_prediction_model_temp_max_binary.onnx
│   └── sunshine_hours_regression_model.onnx
├── rise_imagenet.ipynb
├── rise_mnist.ipynb
├── rise_text.ipynb
└── rise_timeseries_weather.ipynb

----------------------------
## `667-reorganize`  branch
### `dianna` (8 directories, 66 files)
├── __init__.py
├── ...
├── dashboard
│   ├── ...
├── data
│   ├── bee.jpg
│   ├── binary-mnist.npz
│   ├── digit0.jpeg
│   ├── digit0.jpg
│   ├── digit0.png
│   ├── digit1.png
│   ├── leafsnap_example_acer_rubrum.jpg
│   ├── logo.png
│   ├── shapes.npz
│   └── weather_data.npy
├── labels
│   ├── coffee_test.csv
│   ├── coffee_train.csv
│   ├── labels_mnist.txt
│   ├── labels_resnet50.txt
│   ├── labels_text.txt
│   ├── leafsnap_classes.csv
│   ├── movie_reviews_word_vectors.txt
│   └── weather_data_labels.txt
├── methods
│   ├── ...
├── models
│   ├── coffee.onnx
│   ├── geometric_shapes_model.onnx
│   ├── leafsnap_model.onnx
│   ├── mnist_model.onnx
│   ├── mnist_model_tf.onnx
│   ├── movie_review_model.onnx
│   ├── penguin_model.onnx
│   ├── season_prediction_model_temp_max_binary.onnx
│   └── sunshine_hours_regression_model.onnx
├── utils
│   ├── ...
└── visualization
    ├── ...

### `tutorials` (5 directories, 20 files)

├── README.md
├── conversion_onnx
│   ├── keras2onnx.ipynb
│   ├── pytorch2onnx.ipynb
│   ├── skl2onnx.ipynb
│   └── tensorflow2onnx.ipynb
├── demo.ipynb
└── explainers
    ├── KernelSHAP
    │   ├── kernelshap_geometric_shapes.ipynb
    │   ├── kernelshap_mnist.ipynb
    │   ├── kernelshap_tabular_penguin.ipynb
    │   └── kernelshap_tabular_weather.ipynb
    ├── LIME
    │   ├── lime_images.ipynb
    │   ├── lime_tabular_penguin.ipynb
    │   ├── lime_tabular_weather.ipynb
    │   ├── lime_text.ipynb
    │   ├── lime_timeseries_coffee.ipynb
    │   └── lime_timeseries_weather.ipynb
    └── RISE
        ├── rise_imagenet.ipynb
        ├── rise_mnist.ipynb
        ├── rise_text.ipynb
        └── rise_timeseries_weather.ipynb

**Review by loostrum — COMMENTED** [review]
Very nice, thanks for reorganizing this! I very much like that all data files are in one place now. I do have one thing to discuss: we now include everything under `dianna/data` in the pip package, so everyone that install dianna gets a pretty big download as it includes all those files. However, the files are only needed for the tutorials and dashboard. Would it make sense to not include the bigger files in the pip package, and only download them when needed in the notebooks/dashboard code?

**Comment by elboyran** [comment]
> Very nice, thanks for reorganizing this! I very much like that all data files are in one place now. I do have one thing to discuss: we now include everything under `dianna/data` in the pip package, so everyone that install dianna gets a pretty big download as it includes all those files. However, the files are only needed for the tutorials and dashboard. Would it make sense to not include the bigger files in the pip package, and only download them when needed in the notebooks/dashboard code?

You are right. Do you know how to implement this?

**Comment by elboyran** [comment]
@cwmeijer do you mind merging this after the new release?

**Comment by elboyran** [comment]
> > > Very nice, thanks for reorganizing this! I very much like that all data files are in one place now. I do have one thing to discuss: we now include everything under `dianna/data` in the pip package, so everyone that install dianna gets a pretty big download as it includes all those files. However, the files are only needed for the tutorials and dashboard. Would it make sense to not include the bigger files in the pip package, and only download them when needed in the notebooks/dashboard code?
> > 
> > 
> > You are right. Maybe after addressing #171 it will not be an issue.
> 
> Let's leave it for now indeed. For reference: it can be done by editing the MANIFEST.in file. The pypi package default is to _not_ include any non-python files.

Is your review for this PR positive then? ;-)

**Review by loostrum — APPROVED** [review]
Good point, it is! :) 

There are just 2 merge conflicts to resolve first in the notebooks, but I assume that will work out

**Comment by elboyran** [comment]
> Good point, it is! :)
> 
> There are just 2 merge conflicts to resolve first in the notebooks, but I assume that will work out

Indeed, as I mentioned earlier I think these are just 2 notebooks that differ in with or without cell output. I'm not fluent enough in git to resolve manually the conflicts (cannot be done anutomatically). Can someone who knows better, merge the branch 667-reorganise to main? @loostrum, @cwmeijer ?

**Comment by loostrum** [comment]
I'll try fixing it now

**Comment by elboyran** [comment]
[heart]         Elena Ranguelova reacted to your message:
________________________________
